machine_learning/k_means_clust.py

```diff
@@ -202,3 +204,127 @@ def kmeans(
         verbose=True,
     )
     plot_heterogeneity(heterogeneity, k)
+
+
+def ReportGenerator(df, ClusteringVariables, FillMissingReport=None):
```

Comment:
Type hints?  Doctests?

Reply:
Typehints okey. but i cant write doctests. It returns pandas dataframe can you suggest me some ideas how i can do it? We can test manually to be sure that it works

Reply:
Is it impossible to examine various elements of a dataframe to ensure that those elements contain expected values?

Reply:
Its not but i wont be able to check all of them, but i will do good testing with different approaches. Thanks!

Reply:
Just a few sanity checks will be good enough for our purposes.  Thx.